### DISCUSSION TITLE
Beta14 and the new insane plotter!

### DISCUSSION THREAD

**Discussion by loppefaaret** [Chia-Network/chia-blockchain#436] [General]
the 3 new musketeers in the battle of the plots is `-r` `-s` & `-u`...

- `-r` is threads... and its used only during phase 1, table 2-7... 
far from all the work in this phase is able to be run with multiple threads - the work that **can** be, **will** be... but as you can see on some of the reports that are slipping out, it is far from able to run on all threads constantly... 
with my 4core 8thread i7, i was running `-r 8`, and got a total CPU for phase 1 between 230%-250% usage... so, far from 800%...
a run on a 16core 32threads xeon, got it to 353.2% with `-r 32`... my philosophy on this subject is to use atleast `-r` as many cores you got... if you are running parallel plots, and one plotter is already using them, the others are just going to have to share if they all want to use the cores at the same time - if they stagger in between each others usage, then thats just perfect... 
another way to look at it, is to keep 1 core clear for each instance of the plotter, as it at least uses 1 thread... so if you doing 12 parallel plots on a 16 core CPU, i guess a good `-r` choice would be 4 on each...

- `-s` is the amount of cache for each thread - and it defaults to 65536 (64K) if you leave it out - ive tried both 32K and 128K, both at a small loss... most CPU's the last decade has had a L1 cache of 64K, so i think this is the best bet to go with... just dont worry about it, unless you want to take `-s` on an adventure...

- `-u` is the amount of buckets to run the process with - i think of it as datasets... it splits the entire work into pieces of `-u`... `-u` ranges from 16-32-64-128... as far as i understand it, more buckets will cause more random IO, which makes sense, as it works on either 16, 32, 64 or 128 files... as none of us testers have ran a single -t hdd run - its kinda up in the air how its going to perform on HDD's - but any kind of SSD does not care about seeking around in the flash chips...

and this brings me to these 3 new musketeers little helper, their squire... `-b`

- `-b` is now going to be defaulting to 3072 (3G) instead of the previous `-b 2048` (2G) - it gives a good point for gaining speed, and still keeping the memory usage down... its not perfect, but its a pretty good point to go with... i have successfully run `-b 2048` without problems, its just a bit slower... 

_"but why is 2048 slower than 3072"..._ 
its cause of the new brilliant work from the developers, that is now able to mimic running the entire plotting process like one big RAM drive... imagine getting the benefits of 400G of RAM, with just a tad more than 4G...
as you start plotting with the new plotter, you will see 2 different kind of sorting mechanisms in use - uniform sorting, which is the fast, all in memory sorting.... if it is able to fit the entire bucket into the allocated RAM, it will run the entire dataset in one go... if it does not have enough RAM to do so, it will resort to quicksorting - which is kinda like the old way of plotter sorting... not entirely... but somewhat... it will break down the dataset and use the drive while sorting the bucket... 
the slow quicksorting, is still alot faster than any previous versions of the plotter... my tests shows that when you choke the new plotter from using memory, using as little as possible, you still going to see close to twice the speed... 
and if you feed it all the RAM it could ever want, its more than x2 faster... 
my testing is close to concluding that giving the new plotter `-b 4608` MiB of RAM, will allow it to run all the buckets using the faster uniform sorting algorithm, as long as you're maximizing the amount of `-u` buckets to 128.... if you're going the other way about it, and choke it on the amount of -u buckets down to 16... which is an 8th of 128, you need x8 as much ram allocated for it to fully unfold it wings... `-b 36864` (36G)
turning this around, and asking how low you can go with memory - ive concluded a `-u 16 -b 10240` (10G) run, but almost all of the buckets results in the slower quicksorting... if you ran this with `-u 128`, you should be able to lower `-b` to something like 1500 or lower... i personally had a `-b 1274` fail, cause it did not have enough RAM to do quicksort on a bucket... so that plot was wasted... i dont know if a `-b 1280` would be able to complete everytime, as there is some cryptographic variations... so my recommendations for absolute lowest RAM usage so far, is `-b 1500`... the presure point is phase 3, compressing table 5 & 6, second pass... 

oh, and i almost forgot... this new plotter also brings down the amount of writes needed to complete the task... and it doesnt matter if you choke it one way or the other... all my test runs have used just under 2TiB writes... down from 6.6 at worse case... so yippee-ki-yay for that!

disclaimer: im just a google warrior, all of the above is a matter of opinion, subject to bias, and is to be taken only as suggestions!

- `chia plots create -k 32 -b 4608 -u 128 -r 8 -t /SSD -2 /SSD -d /HDD`

**Comment by dorkmo** [comment]
i feel like these new parameters need a 3D chart lol!

very cool, thanks for the details, was a bit overwhelming this morning!

![image](https://user-images.githubusercontent.com/1923070/94963787-469e2a00-04be-11eb-8777-9776f23a8454.png)

**Comment by loppefaaret** [comment]
@scully_chia on keybase has pushed the limits down another 100MB successfully, to 4508
all uniform sorting achieved with `chia plots create -k 32 -u 128 -b 4508 -r 2 -t /SSD -2 /SSD -d /HDD`
and we both have noticed that its not the exact number that you pass onto -b, there is ranges that lands the RAM: value at different plateaus... 
also reports of _very_ decent results on regular HDD plotting even with -u 128... still have no information on just how random the IO is!

**Comment by loppefaaret** [comment]
i somehow managed to ruin it... sweet spot is _not_ 4508.... new target is **4520**... so
`chia plots create -k 32 -u 128 -b 4520 -r 2 -t /ssd -2 /ssd -d /hdd`

**Reply by loppefaaret** [reply]
i have a suspicion that scully_chia found it with -r 4, and i broke it with -r 8... so assigning threads costs a minimal amount of RAM... there is also possibly a small amount of ram that needs added if you're upping -s cache from default 64K

**Comment by loppefaaret** [comment]
basicly, to achieve full uniform sorting on all possible buckets - the first bucket after F1 stage, needs to report higher than `RAM: 3.250GiB` 

if you see this, its good:


> Starting phase 1/4: Forward Propagation into tmp files...
Computing table 1
F1 complete, time: 313.188 seconds. CPU (87.67%)
Computing table 2
	Bucket 0 uniform sort. **_Ram: 3.261GiB_**, u_sort min: 1.125GiB, qs min: 0.281GiB.

on the other hand, if you see this, you going to hit `force_qs: 0` and not get full uniform sort during P1T4&5:
>Starting phase 1/4: Forward Propagation into tmp files...
Computing table 1
F1 complete, time: 317.557 seconds. CPU (88.8%)
Computing table 2
	Bucket 0 uniform sort. **_Ram: 3.242GiB_**, u_sort min: 1.125GiB, qs min: 0.281GiB.

**Comment by andre-bender** [comment]
I am still struggling with my i7-5820k (6 Core). Should I start the first k32 plot with -r 6 and stagger 60 minutes to the next plot or is it better to plot 6 plots with -r 2 each?

**Reply by loppefaaret** [reply]
this was written for the first version of the non-bitfield plotter, back on beta 17 i believe ?
memory req has changed, and other stuff as well, and this post, while somewhat relevant, should be considered outdated

**Comment by HomerChain** [comment]
На что мой компьютер потянет:
Тип ЦП	HexaCore Intel Core i7-8700K, 4300 MHz (43 x 100)
Дисковый накопитель	ST31000528AS  (1 ТБ, 7200 RPM, SATA-II)
Дисковый накопитель	WDC WD2005FBYZ-01YCBB2  (2 ТБ, SATA-III)

Какой именно из этих я смогу запустить и мой комп выдержит и не сгорит ! Подскажите пожалуйста, СПАСИБО!!

101.4GiB (k-32,временное пространство: 239GiB)
208.8GiB (k-33, временное пространство: 521GiB)
429.8GiB (k-34,временное пространство: 1041GiB) 
884.1GiB (k-35,временное пространство: 2175GiB)

**Reply by ilyakooo0** [reply]
Не сгорит он в любом случае. Бутылочное горлышко в большинстве случаев – скорость доступа к временному пространству, а не CPU. Поэтому, в качестве временного пространства лучше использовать SSD. 

Сейчас наиболее целесообразно создавать как можно больше `k-32` плотов.

**Reply by HomerChain** [reply]
> Не сгорит он в любом случае. Бутылочное горлышко в большинстве случаев – скорость доступа к временному пространству, а не CPU. Поэтому, в качестве временного пространства лучше использовать SSD.
> 
> Сейчас наиболее целесообразно создавать как можно больше `k-32` плотов.

спасибо большое

**Comment by PCDSP** [comment]
Why bother... in the last 3 days the TOTAL size of the network has gone up 1.5 times! Just another pyramid scheme... waste of time and energy! While the return is BINGO, lottery... 

P.off..

**Reply by loppefaaret** [reply]
so you made account just to get on a ½ a year old post, and post your frustration ?
how about you join us on keybase - join the community, and perhaps gets some pointers and tips on what you can do to improve your standing in the lottery ? and at some point, the maddness stops... and say that this coin that is 20k times less power hungry than bitcoin, and is said to have a more solid blockchain coding language for dapps than certain others, actually gains some of the traction that it has now... and the price in some years lands where bitcoin is today... so even if you only won 1 time, thats 50k USD... 
atleast just run a single plot, who knows, perhaps you got a free vacation for a week next year ? or a new car in a few years ?

**Reply by PCDSP** [reply]
Why is there 1 user making 30% of the tokens...? 

You're selling it well... so if I plot at over 1TB per 24 hours, do I have any chance, considering the network size just exploded 1.5x times in last 3-4 days!

Dream on... that it will reach 50k / coin!

**Reply by andre-bender** [reply]
Pools are in the making. Chill dude.

**Reply by loppefaaret** [reply]
cause 30% of the total space will get you 30% of the rewards ? sounds fair to me :)
thats about the same speed i do, and im around 80TB now... and i stand to win every 5th day currently, ofcourse, thats about 10% less tomorrow, and so on, but you can also just not do it, and dont have a part in the goldrush... 
and yes, im flying high in my dreams, private jets and everything :)

**Reply by PCDSP** [reply]
Are you winning? As that question mark small print is totally wtf - can take up to 4x times to win! 

Seems to be a ponzy scheme at this pace of network expansion, by the time I reach 100TB, it will say once a month...every 4th month.

**Reply by PCDSP** [reply]
> Pools are in the making. Chill dude.

30% of network = 30% rewards - ok. 

But what about the hype 'you can do this yourself' which probably dragged millions into the hype...

Eco-friendly my arse...90%+ will just waste resources for no return at all...

Join the army of feromone soaked worker ants !

**Reply by loppefaaret** [reply]
yes, i won 12 hours ago
so if you got a 100TB of storage space, that could potentially hold plot files - honestly, what is it you want from chia ? the predictions are sky high, there is solid identifiable people behind, there is a solid corporate structure setup around the premine... the blockchain coding language... the consensus... the much greener aproach to maintaining a blockchain, while keeping the entry as low as possible... there is jsut a so many things that seems to be nailed from block 0... i guess thats why the initial 10k peers is way passed 150k just a little more than a month later, and the diff is climbing like noone in their wildest fantasy would had dared to even joke about... 
just come on in, and join the party please :) im sure you wont regret it

**Reply by loppefaaret** [reply]
and even if 90% is wasting their resources, its still 2000 times greener than bitcoin i guess ?

**Reply by PCDSP** [reply]
Well truly don't know what to do... your example 80TB = 3000€ or more in hard disk space... to run 24/7 to get like 1 XCH / month (calculated within 80 days, if I'm lucky to get 1XCH /mo at the current growth rate)! then failure possibilities of hard drive... It's the 'how you win'... that annoys me as to the length of time involved vs network expansion = mickey mouse, don't want to be a part of a chocolate factory.

PS: Bitcoin is horse $h1t...

**Reply by SackofFluor** [reply]
Well, i started a couple of days ago (when the Network was around 200PiBs), since then I added around 17TiB (my plotting speed is not the greatest).
Since the last two days I am barely able to keep my estimated time to Win around 21-22 days, telling me, that with my plotting speed staying steady and the network accelerating i will start "falling" back.

Taking into consideration how much "stuff" i used which I had laying around, and just adding a little JBOD i don't "sweat" it like others did, But the point still somehow remains valid, the idea (and marketing) about chia allowing the "average joe" to min a little on his Harddrive is not feasible....

Also, we don't have any idea about the true value of the coin since it is not traded so far and I have to admit I don't believe that they will start with 20$.

Edit:
While at the same time we see thousands of people using the GUI farming bloating the network while at the same time most likely don't even farm because the GUI says "farming" but in reality is not checking fast enough or is giving some weird errors/warnings which you only even see once you altered a config file...
So their used hardrive space inflates the network while at the same time only benefits the "masters" of chia, people with huge farms AND the technical knowhow to be 100% sure that their plots are properly counted....

Its kinda weird, that there are unofficial pools with whooping 30% Network power while the dev-team somehow have not found a way to make an "official" pool. For me that sounds "fishy" to put it mildly...

**Reply by loppefaaret** [reply]
we are all loosing ground to the network size now - i dont believe anyone can keep up
it is absolutely madness how fast it is growing - there most be a million computers creating plots
and this cannot continue - withing the next month it has to taper off... (i fear im going to regret this statement)
just bear in mind, that these "unofficial" pools are employing their own closed sources programs needed to be part off
some see them as godsend so they can earn 0.01 chia a day
while others are seeing them as the devil them selfs - forcing closed source programs onto users, inflating their own numbers through deliberate exploiting PB / PIB conversions - under paying their farmers and so on... even if they say they do not have access to the 24 word mnemonics of their farmers - the coins end up on their address - and it is up to them to distribute it out!
just hang in there a bit longer - and the chia team is going to deliver a pooling protocol unlike any other seen previously in crypto!
last i checked, there were 150k+ nodes on the network, and only 11k users on the official keybase chat
somehow the network is closing in on 2EiB - so some of all those machines are doing something right, even if you believe that the chia team is just exploiting them for whatever reason ? 
come on in on keybase - hop on over into #random and lets have a constructive chat about your concerns

**Reply by SackofFluor** [reply]
Na, i dont wanted to insinuate that the Dev-Team is exploiting the users (at least not  to a sizable degree). What I meant with "masters" of chia are the people that have actually really understood the network, make sure that every single plot is always checked properly and that no duplicate plots fill their hard drives etc.

Things that are almost invisible to the "ordinary" GUI-User. Looking at the hard drive prices I have to admit that I don't see the network platooning any time soon. As with all mining, i fear that it will become a Hardware eating monstrosity.

I just hoped that I could get a least a couple of chia before the craze-race would become too absurd, I was wrong :)

Not to mention, it is so much easier to do HDD Mining, especially since we are not depending on super power-hungry GPUs, yet you can put hundred of hard drives in very compact racks...

**Reply by PCDSP** [reply]
There is nothing Eco-Friendly... nor is it realistic - that an average computer user has any chance of earning by keeping the Chia client live. It only helps the total network quantity to release less and less coin... therefore you put your hardware to work for nothing. 

Probably most people fail to grasp the amount of HDD failures that can occur when you have say 200TiB of plots... with the wear and tear on them with the heads seeking every 10seconds or so for plots that can be harvested. All simple words and fun fact, the chia calculator shows a WD external drive - which external drive contains the WD Red, the most prone to failure of the WD lineup, and connected via USB there is a catch - you might find that the seek time is too high for your plot to earn and harvest the XCH...

For HDD farming, you do need Enterprise grade HDDs which are expensive and will not get any cheaper, they might go up up up. Then it's the NVME TBW factor which will cost you a pretty penny. 

Having smaller drives = less failure rate = more watts to keep live 24/7.

The whole earning scheme is grey, and by design only big quantity will earn you anything at this moment in time. 

Simply put, you can lose whatever you invest, unless you already have the hardware sitting around of course... 

And I totally agree, efficient farming of Chia = coding and tweaking the open source. 


I stand by my statement that the whole hype was engineered so that the total network quantity explodes therefore the controlling company has to spread less and less coin to each individual farmer... there is one big earner - Chia Company that will keep the vast majority of the coin. 

Someone in the community last night put a screenshot of 52TiB of plots, farming for the past free weeks with 0 XCH earned, and that is all about the network growth. Bram Cohen said he expects the network to reach 60 EiB end of year 2021. 


LOL! You have to be ignorant and foolish to spend one Watt of juice and computing power into it... but hey, there's plenty around and it will reach 60EiB with some big players continuing to make it big!

**Reply by PCDSP** [reply]
In order to earn anything, you have to keep farming and spending money on HDDs and SSDs non stop!

But how much will it be worth the XCH? Will you lose on your investment? Can you sustain the costs? 

All of the above point to non-average-Joe nor Eco-Friendly...

Put a green leaf logo and a name of a plant, use big words like Eco-Friendly, that will appeal well, they said, over a spliff one night. 'Marketing, checked'

**Reply by andre-bender** [reply]
Chia isn’t fully published until today when transaction starts. So every question can be answered today …

**Reply by PCDSP** [reply]
What questions?
:)

**Reply by andre-bender** [reply]
Are you trolling or what?

those: But how much will it be worth the XCH? Will you lose on your investment? Can you sustain the costs?

**Reply by PCDSP** [reply]
I'm not but here are the answers:

If it will be worth alot, those that have the upper hand of coding (it is open source, liberal) like hpool and others, will invest more. You will be a small pawn that will feed them big, maybe break even...BIG maybe. 

If it will be worth little, the big ones will reduce mining to optimize returns, as others will give up all together. 
Again, you won't be able to justify the expense as the value is too small.

**Reply by PCDSP** [reply]
mining = plotting, to clarify the low value scenario, generating bigger quantity = more storage needed, more costs, they will stick with quota they have and increase it steady not at full throttle... if value is small, to keep costs at bay.

**Reply by PCDSP** [reply]
It's worth a lot! Like 7000€ = 200TiB of enterprise storage in 20 drives of 10TB. To make 1XCH /day NOW, that means... well... fuck'all, as you need to constantly plot to keep up with Netspace expansion! So... Elvis has left the building! 

Anyone argue this is NOT for 95% of computer users, considering the costs vs returns? :)

For sale starter kit :  4TiB of plots in 4 x 1000GB hard disk drives + private key :) 


Returns a pat on the back and a smiley face over time!

I stopped two days ago... :)

6%/week Netspace growth my arse... it's doubled in 7 days!

**Reply by jackykwandesign** [reply]
yep, with the potential i saw in the white paper, and the low electricity, and the manageable  + reversible block-chain under Gov and financial company,  smart contract, its definitely worth something.

So i just throw money to buy machine and HDD when price is low. And even i am not playing the game, only around 20-30% lost for whole investment to resell the HDD, or just a few shipment if i return to Amazon papa. 

So, do use the return policy that benefit to you :)
Nobody will invest a thing with all s/he money

**Reply by ByronAP** [reply]
> Are you winning? As that question mark small print is totally wtf - can take up to 4x times to win!
> 
> Seems to be a ponzy scheme at this pace of network expansion, by the time I reach 100TB, it will say once a month...every 4th month.

I don't think you understand what a "ponzy[sic]" is!

**Comment by padik11** [comment]
Hello.
I have a question. Does the amount of threads allocated in parallel plotting apply to each plot or all?

**Comment by dmialdav** [comment]
Здравствуйте! 
Не могу найти информацию. Какие плоты лучше делать на перспективу? К32? К33? К34?
Одинаков ли доход с плотов К32 и К33?

Hello!
Can't find information. What rafts are better to make in the future? K32? K33? K34?
Is the income from K32 and K33 rafts the same?
